File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& p) {
        int min=p[0];
        int profit=0;
        for(int i=1;i<p.size();i++){
            if(p[i]<min){
                min=p[i];
            }
            else{
                profit=max(profit,p[i]-min);
            }
        } return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] p) {
        int ans = 0;
        int min = p[0];
        for (int i = 1; i < p.length; i++) {
            if (p[i] < min) {
                min = p[i];
            } else {
                ans = Math.max(ans,p[i] - min);
            }
        }
        return ans;
    }
}
